checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -107,5 +115,38 @@ def _handle_static_statement(self, block_index: int) -> Optional[list[str] | dic
             return self._handle_static_count_statement(count_statement)
         return
 
-    def _create_new_foreach_resources(self, block_index: int, foreach_statement: list[str] | dict[str, Any]) -> None:
+    def _handle_dynamic_statement(self, blocks_to_render: list[int]) -> dict[int, Optional[list[str] | dict[str, Any] | int]]:
+        rendered_statements_by_idx: dict[int, Optional[list[str] | dict[str, Any] | int]] = {}
+        sub_graph = self._build_sub_graph(blocks_to_render)
+        self._render_sub_graph(sub_graph, blocks_to_render)
+        for block_idx in blocks_to_render:
+            if not self._is_static_statement(block_idx, sub_graph):
+                rendered_statements_by_idx[block_idx] = None
+            else:
+                rendered_statements_by_idx[block_idx] = self._handle_static_statement(block_idx, sub_graph)
+        return rendered_statements_by_idx
+
+    @staticmethod
+    def _render_sub_graph(sub_graph: TerraformLocalGraph, blocks_to_render: list[int]) -> None:
+        renderer = TerraformVariableRenderer(sub_graph)
+        renderer.vertices_index_to_render = blocks_to_render
+        renderer.render_variables_from_local_graph()
+
+    def _build_sub_graph(self, blocks_to_render: list[int]) -> TerraformLocalGraph:
+        from checkov.terraform.graph_builder.local_graph import TerraformLocalGraph
+        sub_graph = TerraformLocalGraph(self.local_graph.module)
+        sub_graph.vertices = [{}] * len(self.local_graph.vertices)
+        for i, block in enumerate(self.local_graph.vertices):
+            if block.block_type == BlockType.RESOURCE and i not in blocks_to_render:
+                sub_graph.vertices[i] = {}
+            else:
+                sub_graph.vertices[i] = block  # type:ignore
```

Comment:
add a type to your initialization instead of ignoring

Reply:
changed.